src/main/java/seedu/address/storage/JsonAdaptedUid.java

```diff
@@ -35,6 +37,9 @@ public String getUid() {
      * @throws IllegalValueException if there were any data constraints violated in the adapted uid.
      */
     public Uid toModelType() throws IllegalValueException {
+        if (uid == null || Uid.isValidUid(uid)) {
```

Comment:
Hmm not sure if this is the correct logic... I don't think you want an exception thrown if the `uid` is valid!

```suggestion
        if (uid == null || !Uid.isValidUid(uid)) {
```